FlowCrypt/Controllers/Attachment/AttachmentViewController.swift

```diff
@@ -99,22 +131,40 @@ extension AttachmentViewController: WKNavigationDelegate {
     func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
         hideSpinner()
     }
+
+    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction) async -> WKNavigationActionPolicy {
+        guard
+            navigationAction.request.url?.scheme == "data",
+            !isNavigated
+        else { return .cancel }
+
+        isNavigated = true
+        return .allow
+    }
 }
 
 private extension AttachmentViewController {
 
+    private func setContentRules() async throws {
+        guard let list = try await WKContentRuleListStore.default().compileContentRuleList(
+            forIdentifier: "blockRules",
+            encodedContentRuleList: encodedContentRules
+        ) else {
+            throw AppErr.general("Could not produce a data URL to preview this file")
```

Comment:
Should be another error text

Reply:
We can use `do/catch` here and use original error:
```swift
do {
    let list = try await WKContentRuleListStore.default().compileContentRuleList...
    ....
} catch {
    throw error
}
```

Reply:
This will not work here because we get optional and have to log error anyway.